app/src/main/java/com/example/composetemplate/repositories/AuthInteractor.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.composetemplate.repositories
+
+import com.example.composetemplate.data.models.local_models.LoginParameterizable
+import com.example.composetemplate.data.models.local_models.NonSocialLoginParameter
+import com.example.composetemplate.data.models.local_models.User
+import com.example.composetemplate.data.models.server_models.PermissionType
+import com.example.composetemplate.managers.NetworkManager
+import com.example.composetemplate.utils.LoginCallback
+import com.example.composetemplate.utils.extensions.isSuccessful
+import io.ktor.client.statement.HttpResponse
+
+/**
+ *  PRE-CONDITION: enable every auth option in firebase console that you want to use,
+ *  otherwise it will not work and the network calls will go to onFailure
+ *  RECOMMENDATION: Add SHA1 key in firebase project settings, you need it for example, in google auth
+ */
+
+class AuthInteractor(
```

Comment:
I would appreciate it if you could explain the role of this class to me in a Zoom meeting.Like what is the difference for us between an Interactor and a Repository? I see there is a LoginRepository, and this class is in the repositories package.

Reply:
Shon todo: add explaind comment about the Intercator. Main Idea basically its to do more manipulate about logic that you cant do in the repo. like here, I needed to use network class that i Couldnt inject to repo. basically we need to do logic in the smallest layer we can. we can add another interactor package.